### PR TITLE
Backend create endpoint to like a recipe

### PR DESCRIPTION
**Pull request by LenVej**

### CODE REVIEW

File: Backend/Services/UserService.cs [hunk 1]
```diff
@@ -15,10 +15,25 @@ public UserService(IUserRepository repository)
...
+    {
+        var user = _repository.Get(id);
+
+        if (user == null)
+        {
+            return null;
+        }
+
+        return user;
+    }
```
Comment (MaarcS) [inline Backend/Services/UserService.cs:36]: `public User? GetUserById(int id) => _repository.Get(id);` would do the same

File: Backend/Services/UserService.cs [hunk 2]
```diff
@@ -33,6 +48,30 @@ public void LikeRecipe(User user, Recipe recipe)
         user.LikedRecipes.Add(recipe);
         _repository.Update(user);
     }
+    
+    public void UnlikeRecipe(User user, Recipe recipe)
+    {
+        var userWithLikedRecipes = _repository.Get(user.Id);
```
Comment (MaarcS) [inline Backend/Services/UserService.cs:54]: redundant database query

File: Backend/API/Controllers/UserController.cs [hunk 3]
```diff
@@ -7,10 +8,137 @@ namespace API.Controllers;
...
+        if (id == null)
+        {
+            return BadRequest("ID cannot be null.");
+        }
+
+        var user = _userService.GetUserById(id.Value);
+    
+        if (user == null)
+        {
+            return NotFound("Recipe not found.");
```
Comment (MaarcS) [inline Backend/API/Controllers/UserController.cs:56]: wrong error message

**Review by MaarcS — CHANGES_REQUESTED** [review]

File: Backend/Services/UserService.cs (hunk 1 above)
Comment (LenVej) [reply Backend/Services/UserService.cs]: Fixed

File: Backend/Services/UserService.cs (hunk 2 above)
Comment (LenVej) [reply Backend/Services/UserService.cs]: fixed

File: Backend/API/Controllers/UserController.cs (hunk 3 above)
Comment (LenVej) [reply Backend/API/Controllers/UserController.cs]: fixed

**Review by MaarcS — APPROVED** [review]

**Pull request merged**